lib/src/main/kotlin/app/cash/kfsm/State.kt

```diff
@@ -29,6 +29,55 @@ open class State<ID, V : Value<ID, V, S>, S : State<ID, V, S>>(
       .firstOrNull { it.isFailure }
       ?: Result.success(value)
 
+  /**
+   * Finds the shortest path to a given state using a naive recursive algorithm.
+   *
+   * Note: includes this state by default.
+   *
+   * ```kotlin
+   * // Given a simple state machine: [A] -> [B] -> [C]
+   * A.shortestPathTo(C) // listOf(A, B, C)
+   * A.shortestPathTo(A) // listOf(A)
+   * ```
+   *
+   * @param to The state to find a path to.
+   * @return a list of states making up the shortest path.
+   */
+  @Suppress("UNCHECKED_CAST")
```

Comment:
This is necessary because the compiler cannot accurately determine whether `this` is `S`.

Happy to explore better alternatives to this ✌🏻